src/main/java/com/coreos/jetcd/Client.java

```diff
@@ -1,21 +1,24 @@
 package com.coreos.jetcd;
 
+import com.coreos.jetcd.exception.AuthFailedException;
+import com.coreos.jetcd.exception.ConnectException;
+
 /**
  * Etcd Client.
  */
 public interface Client {
 
-  Auth getAuthClient();
+  Auth getAuthClient() throws AuthFailedException, ConnectException;
```

Comment:
I am unsure if throw AuthFailedException ConnectException for all `getAuthClient()`,  `getKVClient()`, etc are user friendly. What if user doesn't want to have lazy client initialization? where the original approach is fine where user expects `AuthFailedException` and  `ConnectException` are thrown at init client build stage and expects subsequent `get*Client` doesn't throw exception.

Two options:
1. make `AuthFailedException` and `ConnectException` unchecked.
2. two client interfaces. The `Client` interface doesn't throw any exception on `get*Client()` and have another interface `LazyClient` that do throw `exception` at `get*Client()`